packages/table/test/loadingOptionsTests.tsx

```diff
@@ -0,0 +1,172 @@
+        ColumnLoadingOption.HEADER,
+    ]);
+    const allRowLoadingOptions = generatePowerSet([
+        RowLoadingOption.CELLS,
+        RowLoadingOption.HEADER,
+    ]);
+
+    afterEach(() => {
+        harness.unmount();
+    });
+
+    after(() => {
+        harness.destroy();
+    });
+
+    /*
+     * Cell loading overrides column loading which in turn overrides table loading. If loading
+     * options are omitted, then the loading options of the parent component are used. Because there
+     * is no `Row` component, row header cell loading overrides table loading. Below is an
+     * exhaustive set of tests of all possible combinations of loading options.
+     */
+    allTableLoadingOptions.forEach((tableLoadingOptions: TableLoadingOption[]) => {
+        allColumnLoadingOptions.forEach((columnLoadingOptions: ColumnLoadingOption[]) => {
+            allRowLoadingOptions.forEach((rowLoadingOptions: RowLoadingOption[]) => {
+                it(`table: ${tableLoadingOptions}, column: ${columnLoadingOptions}, row: ${rowLoadingOptions}`, () => {
+                    const isCellLoading = (index: number) => {
+                        if (index === 0) {
+                            return true;
+                        } else if (index === 1) {
+                            return false;
+                        } else {
+                            return undefined;
+                        }
+                    };
+
+                    const renderCell = (rowIndex: number) => {
+                        return <Cell loading={isCellLoading(rowIndex)}>some cell text</Cell>;
+                    };
+                    const renderColumnHeader = (columnIndex: number) => {
+                        return <ColumnHeaderCell loading={isCellLoading(columnIndex)} name="column header" />;
+                    };
+                    const renderRowHeader = (rowIndex: number) => {
+                        return <RowHeaderCell loading={isCellLoading(rowIndex)} name="row header" />;
+                    };
+
+                    const tableHarness = harness.mount(
+                        <Table loadingOptions={tableLoadingOptions} numRows={2} renderRowHeader={renderRowHeader}>
+                            <Column
+                                loadingOptions={columnLoadingOptions}
+                                renderCell={renderCell}
+                                renderColumnHeader={renderColumnHeader}
+                            />
+                            <Column
+                                loadingOptions={columnLoadingOptions}
+                                renderColumnHeader={renderColumnHeader}
+                            />
+                            <Column
+                                loadingOptions={columnLoadingOptions}
+                                renderColumnHeader={renderColumnHeader}
+                            />
+                        </Table>,
+                    );
```

Comment:
could even define a quick helper component `<TableLoadingOptionsTester>` that does this composition, so it doesn't have to be redefined in each test run.